src/Microsoft.Azure.SignalR.Common/Auth/MicrosoftEntra/AccessKeySynchronizer.cs

```diff
@@ -34,32 +35,50 @@ internal AccessKeySynchronizer(ILoggerFactory loggerFactory, bool start)
         {
             _ = UpdateAccessKeyAsync();
         }
-        _factory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+        _logger = (loggerFactory ?? NullLoggerFactory.Instance).CreateLogger<AccessKeySynchronizer>();
     }
 
-    public void AddServiceEndpoint(ServiceEndpoint endpoint)
+    public Task AddServiceEndpoint(ServiceEndpoint endpoint)
     {
         if (endpoint.AccessKey is MicrosoftEntraAccessKey key)
         {
-            _ = key.UpdateAccessKeyAsync();
+            _keyMap.TryAdd(key, true);
+            return InitializeAccessKeyTask(key);
         }
-        _endpoints.TryAdd(endpoint, null);
+        return Task.CompletedTask;
     }
 
     public void Dispose() => _timer.Stop();
 
     public void UpdateServiceEndpoints(IEnumerable<ServiceEndpoint> endpoints)
     {
-        _endpoints.Clear();
+        _keyMap.Clear();
```

Comment:
UpdateServiceEndpoints does a fresh add? it is possible that some endpoints are already loaded